osu.Game/Screens/Edit/Setup/ResourcesSection.cs

```diff
@@ -46,14 +46,16 @@ private void load()
                 Height = 110,
             };
 
+            bool beatmapHasMultipleDifficulties = working.Value.BeatmapSetInfo.Beatmaps.Count > 1;
+
             Children = new Drawable[]
             {
-                backgroundChooser = new FormFileSelector(".jpg", ".jpeg", ".png")
+                backgroundChooser = new FormBeatmapFileSelector(beatmapHasMultipleDifficulties, ".jpg", ".jpeg", ".png")
                 {
                     Caption = GameplaySettingsStrings.BackgroundHeader,
                     PlaceholderText = EditorSetupStrings.ClickToSelectBackground,
                 },
-                audioTrackChooser = new FormFileSelector(".mp3", ".ogg")
+                audioTrackChooser = new FormBeatmapFileSelector(beatmapHasMultipleDifficulties, ".mp3", ".ogg")
```

Comment:
I'm still of the thought that we should leave such things up to the ranking criteria. I'd rather have a defined flow for this, because if we don't, users will end up documenting weird and potentially incorrect ways of doing it.

Reply:
Well in that case the same thing that applies in https://github.com/ppy/osu/pull/30860#discussion_r1865569965 applies here, because the audio file is *also* encoded to `.osu`.